includes/functions/updates/llms-functions-updates-600.php

```diff
@@ -0,0 +1,204 @@
+ */
+function migrate_achievements() {
+	return _migrate_awards( 'achievement' );
+}
+
+/**
+ * Migrate deprecated meta values for earned certificates.
+ *
+ * @since [version]
+ *
+ * @return bool Returns `true` if more records need to be updated and `false` upon completion.
+ */
+function migrate_certificates() {
+	return _migrate_awards( 'certificate' );
+}
+
+/**
+ * Migrates meta data for achievement and certificate template posts.
+ *
+ * @since [version]
+ *
+ * @return bool Returns `true` if more records need to be updated and `false` upon completion.
+ */
+function migrate_award_templates() {
+
+	$per_page = llms_update_util_get_items_per_page();
+
+	$query = new \WP_Query(
+		array(
+			'orderby'        => array( 'ID' => 'ASC' ),
+			'post_type'      => array( 'llms_achievement', 'llms_certificate' ),
+			'posts_per_page' => $per_page,
+			'no_found_rows'  => true, // We don't care about found rows since we'll run the query as many times as needed anyway.
+			'meta_query'     => array(
+				'relation' => 'OR',
+				array(
+					'key'     => '_llms_achievement_image',
+					'compare' => 'EXISTS',
+				),
+				array(
+					'key'     => '_llms_certificate_image',
+					'compare' => 'EXISTS',
+				),
+			),
+		)
+	);
+
+	foreach ( $query->posts as $post ) {
+		_migrate_image( $post->ID, str_replace( 'llms_', '', $post->post_type ) );
+	}
+
+	// If there was 50 results assume there's another page and run again, otherwise we're done.
+	return ( count( $query->posts ) === $per_page );
+
+}
+
+/**
+ * Update db version to 6.0.0.
+ *
+ * @since [version]
+ *
+ * @return boolean
+ */
+function update_db_version() {
+	\LLMS_Install::update_db_version( '6.0.0' );
+	return false;
+}
+
+/**
+ * Migrate deprecated meta values for user awards by type.
+ *
+ * Queries 50 earned awards at a time and migrates their data by moving meta data
+ * to the new location and then deleting the deprecated meta values.
+ *
+ * @since [version]
+ *
+ * @param string $type Award type, either "achievement" or "certificate".
+ * @return boolean Returns `true` if there are more results and `false` if there are no further results.
+ */
+function _migrate_awards( $type ) {
+
+	$per_page = llms_update_util_get_items_per_page();
+
+	$query = new \WP_Query(
+		array(
+			'orderby'        => array( 'ID' => 'ASC' ),
+			'post_type'      => "llms_my_{$type}",
+			'posts_per_page' => $per_page,
+			'no_found_rows'  => true, // We don't care about found rows since we'll run the query as many times as needed anyway.
+			'fields'         => 'ids', // We just need the ID for the updates we'll perform.
+			'meta_query'     => array(
+				'relation' => 'OR',
+				array(
+					'key'     => "_llms_{$type}_title",
+					'compare' => 'EXISTS',
+				),
+				array(
+					'key'     => "_llms_{$type}_template",
+					'compare' => 'EXISTS',
+				),
+				array(
+					'key'     => "_llms_{$type}_image",
+					'compare' => 'EXISTS',
+				),
+			),
+		)
+	);
```

Comment:
```suggestion
	$query_args = array(
		'orderby'        => array( 'ID' => 'ASC' ),
		'post_type'      => "llms_my_{$type}",
		'posts_per_page' => $per_page,
		'no_found_rows'  => true, // We don't care about found rows since we'll run the query as many times as needed anyway.
		'fields'         => 'ids', // We just need the ID for the updates we'll perform.
		'meta_query'     => array(
			'relation' => 'OR',
			array(
				'key'     => "_llms_{$type}_title",
				'compare' => 'EXISTS',
			),
			array(
				'key'     => "_llms_{$type}_template",
				'compare' => 'EXISTS',
			),
			array(
				'key'     => "_llms_{$type}_image",
				'compare' => 'EXISTS',
			),
		),
	);
	
	if ( 'achievement' === $type ) {
		$query_args['meta_query'][] = array(
			'key'     => '_llms_achievement_content',
			'compare' => 'EXISTS',
		) 
	}
	
	$query = new \WP_Query( $query_args );
```
per https://github.com/gocodebox/lifterlms/pull/1869#issuecomment-977730697